net/unbound/files/unbound.sh

```diff
@@ -375,6 +376,37 @@ unbound_control() {
 
 ##############################################################################
 
+unbound_assistant() {
+  local port=53000
+
+  case "$UB_D_DNS_ASSIST" in
+  ipset-dns)
+    port=$( uci_get ipset-dns.@ipset-dns[0].port )
+
+    if [ ! -f "$UB_ASSIST_CONF" ] \
+    && [ $port -gt 0 ] && [ $port -lt 65535 ] ; then
+      {
+        echo "# $UB_ASSIST_CONF generated by UCI $( date -Is )"
+        echo "forward-zone:"
+        echo "  name: ."
+        echo "  forward-addr: 127.0.0.1@$port"
+        echo "  forward-first: no"
+      } > $UB_ASSIST_CONF
```

Comment:
@EricLuehrsen: Any ideas on how I could customize this `name` part here?

Usually you would only want to forward /some/ DNS queries to `ipset-dns`, so only those IPs are added to an ipset, which can be used as the basis for firewall rules.

For example, here's an example of a config that would add YouTube and Netflix IPs to an ipset in `/etc/config/unbound`:

```
config zone
        option enabled '1'
        option zone_type 'forward_zone'
        option fallback '0'
        list zone_name 'googlevideo.com.'
        list zone_name 'nflxvideo.net.'
        option port '53001'
        list server '127.0.0.1'
```

Reply:
I will need to move the option from being global to per forward zone builder function. First step is the get feature in there and working for exposure, then enhance it. Also `ipset-dns` instances will need to be part of it with LuCI to dynamically help. This was going to be a bit more complicated the first time around. For now, it is the default fall through of other forwards. Depending on what you try to do the negative-logic will be clumsy (awful maybe even), but it works. It depends on if everything is through VPN but a few, or only a few go through VPN.